### PR TITLE
FIR: recursively erase definitely-not-null during override checking

### PR DESCRIPTION
**Pull request by pyos**
And only do this if a language feature for validating definitely-not-null is disabled (< 1.6).

The test for KT-41984 (the issue appears to be access-locked or deleted by the way) otherwise only passes because type enhancement is broken (the type parameter for Inv is incorrectly not enhanced) and this is going to be fixed later.

### CODE REVIEW

File: compiler/testData/ir/irText/types/castsInsideCoroutineInference.kt [hunk 1]
```diff
@@ -1,4 +1,3 @@
-// IGNORE_BACKEND_FIR: JVM_IR
```
Comment (pyos) [inline compiler/testData/ir/irText/types/castsInsideCoroutineInference.kt:1]: This is probably not related to this commit, looks like someone forgot to update the expected data for this test.

**Comment by mglukhikh** [comment]
@pyos , why?

**Comment by pyos** [comment]
It's in #4535. Or did you want to review this separately after all?

**Comment by mglukhikh** [comment]
Ok, understood. No.

**Pull request closed without merging**